torch/csrc/jit/runtime/profiling_record.cpp

```diff
@@ -106,6 +106,7 @@ c10::SymbolicShape ProfilingRecord::mergeSymbolicShapes(
 void ProfilingRecord::insertShapeProfile(Node* n, Value* i) {
   auto pn = createProfileNode(nullptr, {i});
   auto pno = pn->addOutput();
+  pn->ty_(attr::profiled_type, TensorType::get());
```

Comment:
does this simplify some pass so they don't have to check for the existence?
Can we just set it once (the way you are doing below?):

```
        val_type_pair.first->node()->ty_(
            attr::profiled_type, val_type_pair.second);
```

Reply:
```
val_type_pair.first->node()->ty_(attr::profiled_type, val_type_pair.second);
```
only gets invoked for nodes that are run. It does simplify downstream passes to have a consistent representation.